### PR TITLE
imake: add imake/1.0.8 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **imake/1.0.8**

For #6387

dependency tree: [sdl](https://github.com/conan-io/conan-center-index/pull/6387) -> [nas](https://github.com/conan-io/conan-center-index/pull/6388) -> **imake** -> xorg-proto -> xorg-macros

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@jgsogo 
Can you (manually) start a job to create armv8 packages for xorg-proto?

**Comment by jgsogo** [comment]
`xorg-proto` fails for Macos+armv8: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/TapaholesList/232/summary.json

**Comment by jgsogo** [comment]
we are on it: https://github.com/conan-io/conan-center-index/pull/6686

**Comment by madebr** [comment]
This is probably because I added `test_type = "build_requires"` to the test package's recipe.

File: recipes/imake/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
+    def build(self):
+        for src in self.exports_sources:
+            shutil.copy(os.path.join(self.source_folder, src), os.path.join(self.build_folder, src))
+        with self._build_context():
+            self.run("imake", run_environment=True)
```
Comment (madebr) [inline recipes/imake/all/test_package/conanfile.py:40]:
```suggestion
        if not tools.cross_building(self):
            with self._build_context():
                self.run("imake", run_environment=True)
```

File: recipes/imake/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    def test(self):
+        autotools = AutoToolsBuildEnvironment(self)
+        autotools.make()
```
Comment (madebr) [inline recipes/imake/all/test_package/conanfile.py:44]:
```suggestion
        if not tools.cross_building(self):
            autotools = AutoToolsBuildEnvironment(self)
            autotools.make()
```

File: recipes/imake/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+required_conan_version = ">=1.36.0"
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+
+    test_type = "build_requires"
+
+    exports_sources = "Imakefile", "Imake.tmpl"
```
Comment (madebr) [inline recipes/imake/all/test_package/conanfile.py:14]:
```suggestion
    settings = "os", "arch", "compiler", "build_type"

    exports_sources = "Imakefile", "Imake.tmpl"
```

**Comment by madebr** [comment]
@jgsogo 
The error https://github.com/conan-io/conan-center-index/pull/6677#issuecomment-893434563 shows that there is a possible bootstrap problem on c3i when cross building.

File: recipes/imake/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    def build_requirements(self):
+        if not tools.get_env("CONAN_MAKE_PROGRAM"):
+            self.build_requires("make/4.2.1")
+
+    @property
+    def _settings_build(self):
+        return self.settings_build if hasattr(self, "settings_build") else self.settings
```
Comment (madebr) [inline recipes/imake/all/test_package/conanfile.py:20]:
```suggestion
        return getattr(self, "settings_build", self.settings)
```

**Comment by madebr** [comment]
@jgsogo
Weird.
The log contains:
```
    imake/1.0.8:c58972ce7fa79168bcca07b3885d2a61402d8743 - Download
```
Where it should be
```
    imake/1.0.8:c58972ce7fa79168bcca07b3885d2a61402d8743 - Build
```
See https://github.com/conan-io/conan-center-index/pull/6677#issuecomment-893494633

**Comment by jgsogo** [comment]
If you have a look at the [output table](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6677/6-production/imake/1.0.8//summary.json), you can see it says `UPSTREAM + TEST_FAILED` for this configuration: the package is already available in the repository created for this pull-request (it was already built, latest commit is just modifying test_package), and we are just running `conan test` in the CI, so the package needs to be downloaded and the error comes from the `conan test` output, not from `conan install --build`.

The table is useful to know where the error comes from:

image

...although I agree the error message should provide that information as well. Maybe we can just add the first lines of the log output with the command to the snippet we are showing in Github:

```
conan test conan-center-index/recipes/imake/all/test_package/conanfile.py imake/1.0.8@ --profile:host=/Users/jenkins/w/BuildSingleReference/27646/2fdb7b99-e353-499d-aa2f-153a73d71019/profile_host.txt --profile:build=default
```

**Comment by jgsogo** [comment]
Now the actual issue, why package `imake/1.0.8:c58972ce7fa79168bcca07b3885d2a61402d8743` is not found. Probably (this is just a guess) it was being uploaded when the new commit arrived and the previous job was cancelled. Sometimes the cancel request stops gracefully, sometimes it kills the job... if it was uploading the package the upload might be incomplete (some Conan files missing) and Conan thinks the package exists (IIRC it checks `conaninfo.txt`) but at the time it is going to download it, not all the files are available.

Why it doesn't happen more often? Because typically, when someone pushes changes to a PR and the previous job is cancelled, the changes affect the recipe-revision and it doesn't matter if one of the packages is corrupted.

I'll check in Artifactory and (probably) trigger again this PR

**Comment by madebr** [comment]
Thanks.
c3i learning new things always surprises me :smile: 
Great teacher you are :rofl: :wink:

**Comment by jgsogo** [comment]
> Thanks.
> c3i learning new things always surprises me 😄
> Great teacher you are :yoda: 🤣 😉

Thanks! 😊  C3i has some secrets behind the scenes... sometimes it is useful also for me to try to explain the issue, and usually, I mention @danimtb, @uilianries and @SSE4 in case it is something that helps them to improve their understanding of the system (not all of us know about every detail).

---

Indeed, that package-ID is _corrupted_ in Artifactory, it is missing `conanmanifest.txt` file:

image

To fix it, we can just remove the package from that repository, or whatever forces Conan to build again the package and upload it. From the user's perspective, all you can do is force a new recipe revision (or a different PR number).

**Comment by madebr** [comment]
c3i go brrrrrrrrr

**Review by jgsogo — APPROVED** [review]
Step by step 💪

**Review by SSE4 — APPROVED** [review]
paso a paso!

**Comment by madebr** [comment]
stap per stap

**Review by uilianries — APPROVED** [review]

**Pull request merged**